lib/inspec/base_cli.rb

```diff
@@ -120,7 +120,7 @@ def opts
 
     def merged_opts
       # argv overrides json
-      Thor::CoreExt::HashWithIndifferentAccess.new(options_json.merge(options))
+      Thor::CoreExt::HashWithIndifferentAccess.new(options.merge(options_json))
```

Comment:
This will fix the "Thor defaults incorrectly overriding the JSON options" problem, but it introduces the "JSON options overriding user-provided Thor CLI flags" problem. And the comment on line 122 indicates the intention - flags on the CLI should override the JSON.

I think we may have to remove default assignments in Thor and handle them on our own:

* remove all defaults in Thor option declarations
* build a hash of all the options that have defaults, and assign their default values
* merge in the JSON options
* merge in the Thor options